plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -3041,10 +3054,35 @@ public ConnectorMergeTableHandle beginMerge(ConnectorSession session, ConnectorT
 
         beginTransaction(icebergTable);
 
-        IcebergWritableTableHandle insertHandle = newWritableTableHandle(table.getSchemaTableName(), icebergTable, retryMode);
+        List<PositionDeleteFiles> previousDeleteFiles = loadPreviousDeleteFiles(icebergTable);
+        IcebergWritableTableHandle insertHandle = newWritableTableHandle(table.getSchemaTableName(), icebergTable, previousDeleteFiles, retryMode);
         return new IcebergMergeTableHandle(table, insertHandle);
     }
 
+    private static List<PositionDeleteFiles> loadPreviousDeleteFiles(Table icebergTable)
+    {
+        if (formatVersion(icebergTable) < 3) {
```

Comment:
it's effectively == 2, right?

Reply:
Right when this method is called from`beginMerge`. 

However, there is another usage from `beginRefreshMaterializedView`. This method may call `loadPreviousDeleteFiles` even when the format version is 1.